src/Compilers/VisualBasic/Test/Semantic/Diagnostics/GetDiagnosticsTests.vb

```diff
@@ -363,10 +363,10 @@ BC31030: Conditional compilation constant '2' is not valid: Identifier expected.
 ~
 </errors>)
 
-            Assert.Equal(diagnostics(0).Arguments, {"Identifier expected.", "2"})
+            Assert.Equal({ErrorFactory.ErrorInfo(ERRID.ERR_ExpectedIdentifier), "2"}, diagnostics(0).Arguments)
```

Comment:
>{ErrorFactory.ErrorInfo(ERRID.ERR_ExpectedIdentifier), "2"} [](start = 25, length = 59)

This doesn't look right, we should be comparing against base-line in English, otherwise this assert doesn't verify that premature localization doesn't happen. In fact, this Assert is useless, we already used CompilationUtils.AssertTheseDiagnostics to verify the message. Just remove this and other similar asserts

Reply:
On the other hand, it is possible that this assert is used to support the next one in a sense that it ensures we look at the diagnostics object that we want (among the two we have). Given that, it is better to keep the assert, but change it to check only the second argument, that should be enough to identify the object.

---
In reply to: [171931674](https://github.com/dotnet/roslyn/pull/25194#discussion_r171931674) [](ancestors = 171931674)